File: javascript/examples/command-line.js
var rl = require("readline"),
    Rdio = require("./../rdio"),
    cred = require("./rdio_consumer_credentials");

var i = rl.createInterface(process.stdin, process.stdout, null);

function done() {
    i.close();
    process.stdin.destroy();
}

// Create an Rdio object with consumer's credentials.
var rdio = new Rdio([cred.RDIO_CONSUMER_KEY, cred.RDIO_CONSUMER_SECRET]);

// Authenticate against the Rdio service.
rdio.beginAuthentication("oob", function (err, authUrl) {
    if (err) {
        console.log("ERROR: " + err);
        done();
        return;
    }

    console.log("Go to: " + authUrl);

    // Prompt the user for the verifier code.
    i.question("Then enter the code: ", function (verifier) {
        rdio.completeAuthentication(verifier, function (err) {
            if (err) {
                console.log("ERROR: " + err);
                done();
                return;
            }

            // Get a list of playlists.
            rdio.call("getPlaylists", {}, function (err, data) {
                if (err) {
                    console.log("ERROR: " + err);
                    done();
                    return;
                }

                var playlists = data.result.owned;

                console.log("You have the following playlists:");

                playlists.forEach(function (playlist) {
                    console.log(playlist.name);
                });

                done();
            });
        });
    });
});

File: ruby/examples/command-line.rb
#!/usr/bin/env ruby

# (c) 2011 Rdio Inc
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in
# all copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
# THE SOFTWARE.

require 'rubygems'

$LOAD_PATH << '../'
require 'rdio'
require './rdio_consumer_credentials'

# create an instance of the Rdio object with our consumer credentials
rdio = Rdio.new([RDIO_CONSUMER_KEY, RDIO_CONSUMER_SECRET])

# authenticate against the Rdio service
url = rdio.begin_authentication('oob')
puts 'Go to: ' + url
print 'Then enter the code: '
verifier = gets.strip
rdio.complete_authentication(verifier)

# find out what playlists you created
playlists = rdio.call('getPlaylists')['result']['owned']

# list them
playlists.each { |playlist| puts "%s\t%s" % [playlist['shortUrl'], playlist['name']] }
